### PR TITLE
Feature solverhdf5

### PR DESCRIPTION
**Pull request by FFroehlich**

### CODE REVIEW

File: python/amici/__init__.py [hunk 1]
```diff
@@ -223,3 +226,28 @@ def runAmiciSimulations(model, solver, edata_list, failfast=True,
                                                    failfast,
                                                    num_threads)
     return [numpy.ReturnDataView(r) for r in rdata_ptr_list]
+
+
+def readSolverSettingsFromHDF5(*args) -> None:
```
Comment (yannikschaelte) [inline python/amici/__init__.py]: why not explicit arguments? + docstring what is to be passed?

Comment (FFroehlich) [reply python/amici/__init__.py]: Hmm, the function is overloaded (allows passing file either as `str` or `H5::H5File`). In the RTD documentation this is properly processed but looks like tools such as PyCharm do indeed not have proper typhints+docstrings. Will fix as the arguments are the same across overloads.

File: tests/testHDF5.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+
+"""Miscellaneous AMICI Python interface tests"""
```
Comment (dweindl) [inline tests/testHDF5.py]:
```suggestion
"""AMICI HDF5 I/O tests"""
```

File: tests/testHDF5.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+import amici
+import sys
+import copy
+import os
+import unittest
+import importlib
+
+
+class TestAmiciHDF5(unittest.TestCase):
+    """TestCase class various AMICI Python interface functions"""
```
Comment (dweindl) [inline tests/testHDF5.py]:
```suggestion
    """TestCase class for AMICI HDF5 I/O"""
```

File: src/hdf5.cpp [hunk 4]
```diff
@@ -513,6 +513,136 @@ bool attributeExists(H5::H5Object const& object,
...
+    H5LTset_attribute_double(file.getId(), hdf5Location.c_str(),
+                             "atol_fsa", &dbuffer, 1);
+    
+    dbuffer = solver.getRelativeToleranceFSA();
+    H5LTset_attribute_double(file.getId(), hdf5Location.c_str(),
+                             "rtol_fsa", &dbuffer, 1);
+    
+    dbuffer = solver.getAbsoluteToleranceB();
+    H5LTset_attribute_double(file.getId(), hdf5Location.c_str(),
+                             "rtolB", &dbuffer, 1);
```
Comment (dweindl) [inline src/hdf5.cpp]:
```suggestion
                             "atolB", &dbuffer, 1);
```

File: src/hdf5.cpp [hunk 5]
```diff
@@ -513,6 +513,136 @@ bool attributeExists(H5::H5Object const& object,
...
+    
+    dbuffer = solver.getRelativeToleranceB();
+    H5LTset_attribute_double(file.getId(), hdf5Location.c_str(),
+                             "atolB", &dbuffer, 1);
```
Comment (dweindl) [inline src/hdf5.cpp]:
```suggestion
                             "rtolB", &dbuffer, 1);
```

File: src/hdf5.cpp (hunk 4 above)
Comment (FFroehlich) [reply src/hdf5.cpp]: oops

**Pull request merged**